pkg/custompluginmonitor/custom_plugin_monitor.go

```diff
@@ -188,7 +188,7 @@ func (c *customPluginMonitor) generateStatus(result cpmtypes.Result) *types.Stat
 				if condition.Status == types.True && status != types.True {
 					// Scenario 1: Condition status changes from True to False/Unknown
 					newReason = defaultConditionReason
-					if newMessage == "" {
+					if status != types.Unknown {
 						newMessage = defaultConditionMessage
 					} else {
 						newMessage = result.Message
```

Comment:
Add a comment here on why we want to show `result.Message` when it is unknown (as explained in your PR description).